src/Tasks/ManifestUtil/mansign2.cs

```diff
@@ -270,10 +270,14 @@ private void init()
                                Sha256SignatureMethodUri);
 
 #if RUNTIME_TYPE_NETCORE
-            CryptoConfig.AddAlgorithm(typeof(SHA256),
+#pragma warning disable SYSLIB0021 // Type or member is obsolete
+            // SHA256 can not be used since it is an abstract class.
+            // CalculateHashValue internally calls CryptoConfig.CreateFromName and it causes instantiation problems.
+            CryptoConfig.AddAlgorithm(typeof(SHA256Managed),
                                Sha256DigestMethod);
+#pragma warning restore SYSLIB0021 // Type or member is obsolete
```

Comment:
These suppressions bug me. Should we instead

```suggestion
            CryptoConfig.AddOID(Win32.szOID_NIST_sha256,
                               Sha256DigestMethod);
```

?

Reply:
According to the finder, Reference.CalculateHashValue ends up calling CryptoConfig.CreateFromName to create the Algorith object. From what I can tell, CreateFromName uses the mapping written to by AddAlgorithm [appNameHT dict] to create the Algorithm object. The AddOID writes to appOidHT mapping table which does not appear to be used by CreateFromName.

It would be good to try and confirm this though.

Reply:
It didn't work, I have just noticed it.
Falling back to 
```
CryptoConfig.AddAlgorithm(typeof(SHA256Managed),	
                               Sha256DigestMethod); 
```
usage

Reply:
@rainersigwald fyi